code/game/gamemodes/objective.dm

```diff
@@ -845,15 +845,22 @@ GLOBAL_LIST_EMPTY(possible_items_special)
 		var/list/all_items = M.current.get_all_contents() //this should get things in cheesewheels, books, etc.
 		for(var/obj/I in all_items) //Check for wanted items
 			if(is_type_in_typecache(I, wanted_items))
-				stolen_count++
+				if(check_if_valid_item(I))
+					stolen_count++
 	return stolen_count >= amount
 
+/datum/objective/steal_n_of_type/proc/check_if_valid_item(obj/item/I)
+	return TRUE
+
 /datum/objective/steal_n_of_type/summon_guns
 	name = "steal guns"
 	explanation_text = "Steal at least five guns!"
 	wanted_items = list(/obj/item/gun)
 	amount = 5
 
+/datum/objective/steal_n_of_type/summon_guns/check_if_valid_item(obj/item/gun/G)
```

Comment:
Keep the old variable name and type, we shouldn't change these very often.

Then check and cast inside the proc body itself.

Reply:
done